.vib/nats/goss/goss.yaml

```
@@ -0,0 +1,28 @@
+file:
+  /bitnami/nats/conf/{{ .Vars.natsFilename }}.conf:
+    mode: "0644"
+    filetype: file
+    exists: true
+    owner: root
+command:
+  {{- $subject := printf "subj_%s" (randAlpha 5) }}
+  {{- $msg := printf "msg_%s" (randAlpha 5) }}
+  {{- $auth := printf "--user=%s --password=%s" .Vars.auth.user .Vars.auth.password }}
+  {{- $client_endpoint := printf "-s nats://nats:%d" .Vars.service.ports.client }}
+  manage-key:
+    ## We need to have the subscriber process running at the same time the publishing is done
+    ## The sleep is needed to avoid a race condition where the publishing is done before the subscriber is ready
+    ## The subscriber process needs to be killed so there is no dangling process making goss timeout
+    exec: ( USER=1001 nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
```

Use the same user as in securityContext

```suggestion
    exec: ( USER={{ .Vars.containerSecurityContext.runAsUser }} nats {{ $auth }} {{ $client_endpoint }} sub {{ $subject }}; ) & sleep 3 && USER=1001 nats {{ $auth }} {{ $client_endpoint }} pub {{ $subject }} {{ $msg }} && sleep 3 && pkill -f 'nats '
```